test/parallel/test-worker-fshandles-closed.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs/promises');
+const {
+  setImmediate,
+} = require('timers/promises');
+const { isMainThread, parentPort, Worker } = require('worker_threads');
+
+const MAX_ITERATIONS = 20;
+const MAX_THREADS = 10;
+
+if (isMainThread) {
+  function spinWorker(iter) {
+    const w = new Worker(__filename, { workerData: iter });
+    w.on('message', common.mustCall((msg) => {
+      assert.strictEqual(msg, 'terminate');
+      w.terminate();
+    }));
+
+    w.on('exit', common.mustCall(() => {
+      if (iter < MAX_ITERATIONS)
+        spinWorker(++iter);
+    }));
+  }
+
+  for (let i = 0; i < MAX_THREADS; i++) {
+    spinWorker(0);
+  }
+} else {
+  async function open_close_ok() {
+    const fh = await fs.open(__filename);
+    await fh.close();
+    await setImmediate();
+    await open_close_ok();
+  }
+
+  async function open_close_nok() {
+    await assert.rejects(
+      fs.open('this file does not exist'),
+      {
+        code: 'ENOENT',
+        syscall: 'open'
+      }
+    );
+    await setImmediate();
+    await open_close_nok();
+  }
+
+  open_close_ok().then(common.mustCall());
+  open_close_nok().then(common.mustCall());
```

Comment:
```suggestion
  open_close_ok();
  open_close_nok();
```
The then callbacks are never actually run. Replacing with `mustNotCall()` passes too.

Reply:
If we remove the `.then`, we should probably add a comment explaining why.